### PR TITLE
CP-387: Firefox tweaks for MV3 support

### PR DESCRIPTION
**Pull request by GabrielCousin**
## 📝 What is this PR

-   [Specs](https://phantombuster.atlassian.net/browse/CP-387)

Weeeeeeell, it was some epic debugging times 🔫 
This addresses two kind of things:
- Firefox and Chrome don't have the same strict specification for `manifest` 
- The extension was making all Firefox tab crash

### Packaging the manifest for Firefox
The `background.service_worker` property is not supported at all https://extensionworkshop.com/documentation/develop/manifest-v3-migration-guide/ I created a script that will patch manifest.json for Firefox (it's very close to the previous `background.scripts` v2 notation except you must not use the `persistent` property

Here are the new steps:
```sh
$ npm run build
$ npm run patch:firefox
$ npm run zip:beta
```

### Other issues
After packaging successfully, the extension was making all tabs blowing up in Firefox and it was hard to debug (no errors). I finally managed to narrow the scope and noticed it was related to Sentry:
- I checked if it could be related to calling `init` several times but scopes and contexts are expected to be different
- At the end I managed to recreate what the `init` method from `@sentry/browser` was doing and tested options one by one (I said it was a problem with `fetch` but it's not exactly that, the problem comes from the code in charge of checking fetch support) 👉 this is solved by forcing the `transport` option
- After the previous step, the tab was finally starting but the extension throw some errors such as `addEventListener is read-only` - in that case it was easier to debug, and I tweaked the integration settings so it finally works

### Build change

I made another change in the build: I noticed the manifest was imported and I wrongly assume keeping it into the build could affect Sentry and sourcemaps (at the end there will be one build per version, so one sourcemaps per version but one manifest per browser) - anyway, it was wrong but I decided to keep the change so it's not importing the (default/Chrome) manifest anymore (`webpack` is taking care of inject environment variable and replace the version), saving 1Kb 🙄 - I can revert the change if you prefer

## 🗒 Notes

⚠️ I don't know if we've experienced this in the past but it seems that in Firefox you have to opt-in website by website so an extension can access data all the time ❓ 

### ✅ Positive testing (should succeed)

- [ ] Install in both browser
- [ ] The extension banner must disapear
- [ ] The Linkedin cookie must be filled

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -44,6 +45,9 @@ module.exports = {
 	},
 	plugins: [
 		...envPlugins,
+		new EnvironmentPlugin({
```
Comment (RiahiKarim) [inline webpack.config.js:48]: TIL. This is neat. I was seeing `process.env` in a browser context (now I understand 😃).

**Comment by GabrielCousin** [comment]
Regarding the notes:
https://blog.mozilla.org/addons/2022/05/18/manifest-v3-in-firefox-recap-next-steps/
Screenshot 2023-03-03 at 15 55 46
#craughing

**Review by LoganPB — APPROVED** [review]
LGTM.

The V3 in Firefox is not user friendly...
image

We'll have to think how we can manage that

**Review by RiahiKarim — COMMENTED** [review]
I think that we should hold off merging this PR. The new "User controls for site access" will break disable ("break") the extension for many users.

**Comment by GabrielCousin** [comment]
@LoganPB @RiahiKarim @Hatscat considering the most critical feedback here (the described in "Notes") - I see three options:
- We don't merge the PR (the probably is that we will have to live with 2 different versions)
- We keep it as is (which will have bad impact on users: they will need to tweak their settings but that's going to be a standard "by Firefox design" - they aimed at deprecating v2 by EOY 2023)
- Let's maintain 2 manifests (v3 for Chrome, v2 for Firefox) - I would require a small effort in documentation and automation but we can reuse the code with no problem. Note that even if we chose the option 2, the manifest has to be patched. (this is my preferred approach to be honest, lowest maintenance effort on the codebase itself, and we have time to analyse how Firefox will evolve with mv3)

What do you think? We can have a formal chat if you prefer 🎙️

**Review by Hatscat — APPROVED** [review]
LGTM 💯

**Review by RiahiKarim — APPROVED** [review]
LGTM 💯

**Pull request merged**